yarn-project/simulator/src/avm/avm_simulator.ts

```diff
@@ -60,6 +71,7 @@ export class AvmSimulator {
     isStaticCall: boolean,
     calldata: Fr[],
     allocatedGas: Gas,
+    instructionSet: InstructionSet,
```

Comment:
This is the only change that "i don't love". 
I see you are constructing this instruction set at the TX level as a private member. Could we just do it here and then we don't change the interface? (and also then don't need to change anything in the tx simulator and processor)

I realize that would still re-create it on each nested call but maybe that's not so bad (compared to the previous state).

Alternatively, you could maybe have a static variable in AvmSimulator for it which is `InstructionSet | undefined`, then initialize it inside `create` if undefined. Not sure if worth the complication though.

Reply:
That seems reasonable.

Reply:
This is done